libs/cluster/Server/Migration/MigrateSessionKeys.cs

```diff
@@ -68,25 +67,16 @@ private bool MigrateKeysFromMainStore()
                         continue;
                     }
 
-                    // Make value SpanByte
-                    SpanByte value;
-                    MemoryHandle memoryHandle = default;
+                    // Get SpanByte, check expiration and write to network buffer
+                    ref var value = ref o.SpanByte;
                     if (!o.IsSpanByte)
-                    {
-                        memoryHandle = o.Memory.Memory.Pin();
-                        value = SpanByte.FromPinnedMemory(o.Memory.Memory);
-                    }
-                    else
-                        value = o.SpanByte;
+                        value = ref SpanByte.ReinterpretWithoutLength(o.Memory.Memory.Span);
 
                     if (!ClusterSession.Expired(ref value) && !WriteOrSendMainStoreKeyValuePair(ref key, ref value))
                         return false;
```

Comment:
Make sure all returning code paths dispose the memory here and in other places.

Reply:
Good catch! I added a dispose call when write to network buffer fails

Reply:
Try-finally might be a more robust pattern of were know this is a batched operation.

Reply:
Check 3b08922 commit. I also re-use the heap buffer across multiple reads now. So, we can avoid multiple allocations if the rented buffer is big enough for the value we are reading.